pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -237,6 +237,21 @@ func redeployAgent(cluster *v3.Cluster, desiredAgent, desiredAuth string, desire
 		return true
 	}
 
+	agentEnvVarsChanged := false
+	desiredAgentEnvVars := cluster.Spec.AgentEnvVars
+	currentAgentEnvVars := cluster.Status.AppliedAgentEnvVars
+
+	if (desiredAgentEnvVars == nil && currentAgentEnvVars != nil) || (desiredAgentEnvVars != nil && currentAgentEnvVars == nil) {
```

Comment:
I think `reflect.DeepEqual(desiredAgentEnvVars, currentAgentEnvVars)` should handle all the cases?

Reply:
So this can be 
```
if !reflect.DeepEqual(cluster.Spec.AgentEnvVars, cluster.Status.AppliedAgentEnvVars) {
    logrus.Infof("clusterDeploy: redeployAgent: redeploy Rancher agents due to agent env vars mismatched for [%s], was [%v] and will be [%v]", cluster.Name, currentAgentEnvVars, desiredAgentEnvVars)
    return true
}
```